### PR TITLE
no-unreachable: allow type alias after function return

### LINKED ISSUE #564: Incorrect unreachable code error for type alias after function return
```ts
function foo() {
  return 1;
  function bar() {}
  type Bar = string;
}
```
Reports unreachable code on the type alias. 

Using tslint v2.5.0-dev.1

### PR DESCRIPTION
**Pull request by YuichiNukiyama**
fix #564

### CODE REVIEW

**Comment by adidahiya** [comment]
thanks for the PR!

lgtm 👍

File: src/rules/noUnreachableRule.ts [hunk 1]
```diff
@@ -99,4 +100,8 @@ class NoUnreachableWalker extends Lint.RuleWalker {
         super.visitThrowStatement(node);
         this.hasReturned = true;
     }
+
+    public visitTypeAliasDeclaration(node: ts.TypeAliasDeclaration) {
```
Comment (JKillian) [inline src/rules/noUnreachableRule.ts]: I don't think we actually need to add these lines - `visitNode` gets called on all nodes by default

File: src/rules/noUnreachableRule.ts [hunk 2]
```diff
@@ -36,8 +36,9 @@ class NoUnreachableWalker extends Lint.RuleWalker {
 
     public visitNode(node: ts.Node) {
         const previousReturned = this.hasReturned;
-        // function declarations can be hoisted -- so set hasReturned to false until we're done with the function
-        if (node.kind === ts.SyntaxKind.FunctionDeclaration) {
+        // function declarations and type alias declarations can be hoisted
+        // -- so set hasReturned to false until we're done with the function
+        if (node.kind === ts.SyntaxKind.FunctionDeclaration || node.kind === ts.SyntaxKind.TypeAliasDeclaration) {
```
Comment (JKillian) [inline src/rules/noUnreachableRule.ts]: Down in line 54 we should restore the state of `this.hasReturned` like we do for function declarations

**Comment by JKillian** [comment]
Hi @YuichiNukiyama, thanks for the PR, just a couple tweaks and then we can merge this!

**Comment by YuichiNukiyama** [comment]
Thank you for your review. I did it.

**Comment by JKillian** [comment]
Looks great, thanks!

**Pull request merged**